cpp/oneapi/dal/algo/knn/infer_types.cpp

```diff
@@ -70,6 +72,10 @@ infer_result<Task>::infer_result() : impl_(new infer_result_impl<Task>{}) {}
 
 template <typename Task>
 const table& infer_result<Task>::get_responses() const {
+    using msg = dal::detail::error_messages;
+    if (!bool(get_result_options() & result_options::responses)) {
```

Comment:
Suggest to introduce the `.test(...)` method to the result options:
```c++
if (get_result_options().test(result_options::responses))
```

Reply:
won't fix

Reply:
result options should behave like ordinary enum. It looks like such functionality violates this rule

Reply:
This is disadvantage of enum, if we have opportunity to fix this limitation via classes, why not?
It violates nothing, adding this method does not break enum-like behavior.